test/router-luatest/router_test.lua

```diff
@@ -285,6 +285,133 @@ g.test_map_callrw_raw = function(g)
     end)
 end
 
+g.test_group_buckets = function(g)
+    local res = g.router:exec(function()
+        local val, err = ivshard.router.group({2, 1, 1})
+        return {
+            val = val,
+            err = err,
+        }
+    end)
+    assert(res.err == nil)
+    local rs2_uuid = g.replica_2_a:replicaset_uuid()
+    local expected = {
+        [rs2_uuid] = {1, 2},
+    }
+    t.assert_equals(res.val, expected)
+end
+
+g.test_partial_callrw = function(g)
+    local expected, rs2_uuid, res
+
+    local create_map_func_f = function(res1)
+        rawset(_G, 'do_map', function(res2)
+            return {res1, res2}
+        end)
+    end
+    g.replica_1_a:exec(create_map_func_f, {1})
+    g.replica_2_a:exec(create_map_func_f, {2}) 
+
+    --
+    -- Successful partial map.
+    --
+    local res = g.router:exec(function()
+        local val, err, err_uuid = ivshard.router.partial_callrw({1, 2}, 'do_map', {3},
+            {timeout = iwait_timeout})
+        return {
+            val = val,
+            err = err,
+            err_uuid = err_uuid,
+        }
+    end)
+    t.assert(not res.err)
+    t.assert(not res.err_uuid)
+    rs2_uuid = g.replica_2_a:replicaset_uuid()
+    expected = {
+        [rs2_uuid] = {{2, 3}},
+    }
+    t.assert_equals(res.val, expected, 'partial callrw success')
```

Comment:
In all your tests you only cover call on a single storage. Please, cover the case when at least 2 storages are used.

Reply:
Implemented additional test case for multiple storages